File: src/main/java/org/home/todobackend/controller/PriorityController.java
package org.home.todobackend.controller;
import org.home.todobackend.entity.Priority;
import org.home.todobackend.search.PrioritySearchValues;
import org.home.todobackend.service.PriorityService;
import org.springframework.dao.EmptyResultDataAccessException;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.NoSuchElementException;

/*
Используем @RestController вместо обычного @Controller, чтобы все ответы сразу оборачивались в JSON,
иначе пришлось бы добавлять лишние объекты в код, использовать @ResponseBody для ответа, указывать тип отправки JSON
Названия методов могут быть любыми, главное не дублировать их имена внутри класса и URL mapping
*/

@RestController
@RequestMapping("/priority") // базовый URI
public class PriorityController {
    // доступ к данным из БД
    private PriorityService priorityService;

    // автоматическое внедрение экземпляра класса через конструктор
    // не используем @Autowired ля переменной класса, т.к. "Field injection is not recommended "
    public PriorityController(PriorityService priorityService) {
        this.priorityService = priorityService;
    }

//    @GetMapping("/id")
//    public Priority findById() {
//        return PriorityService.findById(120127L);
//    }

    @PostMapping("/all")
    public List<Priority> findAll(@RequestBody String email) {
        return priorityService.findAll(email);
    }

    @PostMapping("/add")
    public ResponseEntity<Priority> add(@RequestBody Priority priority) {
        //проверка на обязательные параметры
        if (priority.getId() != null && priority.getId() != 0) { //Уже не новый объект
            return new ResponseEntity("redundant param: id mast be null!", HttpStatus.NOT_ACCEPTABLE);
        }
        //если передали пустые значимые поля
        if (priority.getTitle() == null || priority.getTitle().trim().length() == 0) {
            return new ResponseEntity("missed param: title mast be not null!", HttpStatus.NOT_ACCEPTABLE);
        }
        return ResponseEntity.ok(priorityService.add(priority));
    }

    @PutMapping("/update")
    public ResponseEntity update(@RequestBody Priority priority) {
        //проверка на обязательные параметры
        if (priority.getId() == null || priority.getId() == 0) { //Еще новый объект
            return new ResponseEntity("missed param: id mast be not null!", HttpStatus.NOT_ACCEPTABLE);
        }
        //если передали пустые значимые поля
        if (priority.getTitle() == null || priority.getTitle().trim().length() == 0) {
            return new ResponseEntity("missed param: title mast be not null!", HttpStatus.NOT_ACCEPTABLE);
        }
        priorityService.update(priority);
        return new ResponseEntity(HttpStatus.OK);
    }

    //    @DeleteMapping("/delete/{id}/{email}")
//    public ResponseEntity delete(@PathVariable("id") Long id, @PathVariable("email") String email) {
    @DeleteMapping("/delete/{id}")
    public ResponseEntity delete(@PathVariable("id") Long id) {
        try {
            priorityService.delete(id);
        } catch (EmptyResultDataAccessException ex) {
            ex.printStackTrace();
            return new ResponseEntity("id = "+id+" not found!", HttpStatus.NOT_ACCEPTABLE);
        }
        return new ResponseEntity(HttpStatus.OK);
    }

    @PostMapping("/id")
    public ResponseEntity<Priority> getByID(@RequestBody Long id) {
        Priority priority = null;
        try {
            priority = priorityService.findById(id);
        } catch (NoSuchElementException ex) {
            ex.printStackTrace();
            return new ResponseEntity("id = "+id+" not found!", HttpStatus.NOT_ACCEPTABLE);
        }
        return ResponseEntity.ok(priority);
    }

    @PostMapping("/search")
    public ResponseEntity<List<Priority>> search(@RequestBody PrioritySearchValues prioritySearchValues) {
        if (prioritySearchValues.getEmail() == null || prioritySearchValues.getEmail().trim().length() == 0) {
            return new ResponseEntity("missed param: email mast be not null!", HttpStatus.NOT_ACCEPTABLE);
        }
        List<Priority> priorities = priorityService.findByTitle(prioritySearchValues.getTitle(), prioritySearchValues.getEmail());
        return ResponseEntity.ok(priorities);
    }
    
}

File: src/main/java/org/home/todobackend/controller/PriorityController.kt
package org.home.todobackend.controller

import org.home.todobackend.entity.Priority
import org.home.todobackend.search.SearchValues
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.RequestMapping
import org.home.todobackend.service.PriorityService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.http.ResponseEntity
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.dao.EmptyResultDataAccessException
import java.util.NoSuchElementException


/*
Используем @RestController вместо обычного @Controller, чтобы все ответы сразу оборачивались в JSON,
иначе пришлось бы добавлять лишние объекты в код, использовать @ResponseBody для ответа, указывать тип отправки JSON
Названия методов могут быть любыми, главное не дублировать их имена внутри класса и URL mapping
*/
@RestController
@RequestMapping("/priority") // базовый URI
// автоматическое внедрение экземпляра класса через конструктор
// не используем @Autowired ля переменной класса, т.к. "Field injection is not recommended "
class PriorityController (private val priorityService: PriorityService) {

    //    @GetMapping("/id")
    //    public Priority findById() {
    //        return PriorityService.findById(120127L);
    //    }
    @PostMapping("/all")
    fun findAll(@RequestBody email: String): List<Priority> {
        return priorityService.findAll(email)
    }

    @PostMapping("/add")
    fun add(@RequestBody priority: Priority): ResponseEntity<Any> {
        //проверка на обязательные параметры
        if (priority.id != null && priority.id != 0L) { //Уже не новый объект
            return ResponseEntity<Any>("redundant param: id mast be null!", HttpStatus.NOT_ACCEPTABLE)
        }
        //если передали пустые значимые поля
        if (priority.title == null || priority.title.trim().isEmpty()) {
            return ResponseEntity<Any>("missed param: title mast be not null!", HttpStatus.NOT_ACCEPTABLE)
        }
        return ResponseEntity.ok(priorityService.add(priority))
    }

    @PutMapping("/update")
    fun update(@RequestBody priority: Priority): ResponseEntity<Any> {
        //проверка на обязательные параметры
        if (priority.id == null || priority.id == 0L) { //Еще новый объект
            return ResponseEntity<Any>("missed param: id mast be not null!", HttpStatus.NOT_ACCEPTABLE)
        }
        //если передали пустые значимые поля
        if (priority.title == null || priority.title.trim().isEmpty()) {
            return ResponseEntity<Any>("missed param: title mast be not null!", HttpStatus.NOT_ACCEPTABLE)
        }
        priorityService.update(priority)
        return ResponseEntity<Any>(HttpStatus.OK)
    }

    //    @DeleteMapping("/delete/{id}/{email}")
    //    public ResponseEntity delete(@PathVariable("id") Long id, @PathVariable("email") String email) {
    @DeleteMapping("/delete/{id}")
    fun delete(@PathVariable("id") id: Long): ResponseEntity<Any> {
        try {
            priorityService.delete(id)
        } catch (ex: EmptyResultDataAccessException) {
            ex.printStackTrace()
            return ResponseEntity<Any>("id = $id not found!", HttpStatus.NOT_ACCEPTABLE)
        }
        return ResponseEntity<Any>(HttpStatus.OK)
    }

    @PostMapping("/id")
    fun getByID(@RequestBody id: Long): ResponseEntity<Any> {
        var priority: Priority = try {
            priorityService.findById(id)
        } catch (ex: NoSuchElementException) {
            ex.printStackTrace()
            return ResponseEntity<Any>("id = $id not found!", HttpStatus.NOT_ACCEPTABLE)
        }
        return ResponseEntity.ok(priority)
    }

    @PostMapping("/search")
    fun search(@RequestBody prioritySearchValues: SearchValues): ResponseEntity<Any> {
        if (prioritySearchValues.email == null || prioritySearchValues.email.trim().isEmpty()) {
            return ResponseEntity<Any>("missed param: email mast be not null!", HttpStatus.NOT_ACCEPTABLE)
        }
        val priorities = priorityService.findByTitle(prioritySearchValues.title, prioritySearchValues.email)
        return ResponseEntity.ok(priorities)
    }
}